### PR TITLE
navbarElement and weekdayElement instead of component

### PR DESCRIPTION
**Pull request by boatkorachal**
from #177

### CODE REVIEW

**Comment by gpbl** [comment]
Hi @boatkorachal thanks for the help and congrats for your first PR 🎊 

To maintain backward compatibility, we _should not_ remove these `*Component` props. My idea was  actually to _add_ two new props. We could eventually deprecate them. Sorry for not being clear enough: I'll add some comments to your PR.

**Comment by gpbl** [comment]
(You should rebase your branch with master to fix these conflicts)

File: src/DayPicker.js [hunk 1]
```diff
@@ -43,8 +43,8 @@ export default class DayPicker extends Component {
     onCaptionClick: PropTypes.func,
 
     renderDay: PropTypes.func,
-    weekdayComponent: PropTypes.func,
-    navbarComponent: PropTypes.func,
```
Comment (gpbl) [inline src/DayPicker.js]: Keep those lat two lines. I'll add a PropType to deprecate them.

File: src/DayPicker.js [hunk 2]
```diff
@@ -63,8 +63,8 @@ export default class DayPicker extends Component {
     canChangeMonth: true,
     reverseMonths: false,
     renderDay: day => day.getDate(),
-    weekdayComponent: Weekday,
-    navbarComponent: Navbar,
```
Comment (gpbl) [inline src/DayPicker.js]: idem

File: src/DayPicker.js [hunk 3]
```diff
@@ -328,7 +328,29 @@ export default class DayPicker extends Component {
...
+      previousMonth: this.getPreviousNavigableMonth(),
+      showPreviousButton: this.allowPreviousMonth(),
+      showNextButton: this.allowNextMonth(),
+      onNextClick: this.showNextMonth,
+      onPreviousClick: this.showPreviousMonth,
+      dir: attributes.dir,
+      locale,
+      localeUtils,
+    };
+    return React.cloneElement(navbarElement, props);
```
Comment (gpbl) [inline src/DayPicker.js]: Here we should write instead

```
if (navbarComponent) {
  return React.createElement(navbarComponent, props);
}
return React.cloneElement(navbarElement, props);
```

**Comment by boatkorachal** [comment]
ewww i just did the rebase and force push. hope that i haven't done anything wrong :)

**Comment by gpbl** [comment]
Yay thanks! Could you even test that the rendered component is actually the navbarElement? 🙏 
https://coveralls.io/builds/6541993/source?filename=src%2FDayPicker.js#L357

**Comment by boatkorachal** [comment]
Isn't it checked at proptypes? or should I import react-addons-test-utils and change to something like this

``` javascript
if ( TestUtils.isElement(navbarElement) ) {
  return React.cloneElement(navbarElement, props)
}
```

**Comment by gpbl** [comment]
The test coverage has been decreased, this is because we didn't include a test to check if passing a `navbarElement` did actually render the element. See https://coveralls.io/builds/6541993/source?filename=src%2FDayPicker.js#L357 

Actually, even `weekdayElement` hasn't be checked: I wonder we are actually rendering it :) 
I can't work on this now, so that's why I'm asking :)

**Comment by boatkorachal** [comment]
I have just update tests. Is it ok? (I have to admit that this is the first time i write the tests too haha)

**Comment by boatkorachal** [comment]
oh sorry i forgot to run lint. i'm gonna fix it now.

**Comment by gpbl** [comment]
Looks good! Thanks a lot!

**Pull request merged**